pagopa/io-web-profile
# [IOPID-660] : Adding Retry Fetch

## Short description

Adding the functionality for managing multiple retries in case of timeout (defined threshold) or received specifica status code.

## List of changes proposed in this pull request

- Fetch Multiple Retries
- Adding API Fetch Param to ENV File

## How to test

Start the project, set the environment variables, for example as follow:

NEXT_PUBLIC_API_FETCH_TIMEOUT=5000
NEXT_PUBLIC_API_FETCH_MAX_RETRY=5

In Mockoon, set the delay time to 6000 for the "/profile" endpoint. 

Perform the log-in, verify that the /profile endpoint is called 5 times, as it exceeds the timeout limit.
Repeat the same operation by setting the response for the "/profile" endpoint to 500.
